### PR TITLE
f

### PR DESCRIPTION
**Pull request by XOctOpus1**

### CODE REVIEW

**Review by vsmutok — CHANGES_REQUESTED** [review]
what does this `f` mean?
please create normal commit names

image

File: services/order.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+from db.models import User, Ticket, Order, MovieSession
+
+
+def create_order(
+        tickets: list[Ticket],
+        username: str,
+        date: str = None
+) -> Order:
+
+    with transaction.atomic():
```
Comment (vsmutok) [inline services/order.py:12]: You could use the transaction.atomic decorator for this method.

File: services/movie.py [hunk 2]
```diff
@@ -28,13 +32,14 @@ def create_movie(
...
-        title=movie_title,
-        description=movie_description,
-    )
-    if genres_ids:
-        movie.genres.set(genres_ids)
-    if actors_ids:
-        movie.actors.set(actors_ids)
-
-    return movie
+    with transaction.atomic():
```
Comment (vsmutok) [inline services/movie.py:35]: You could use the transaction.atomic decorator for this method.

**Review by vsmutok — CHANGES_REQUESTED** [review]

File: db/models.py [hunk 3]
```diff
@@ -50,3 +59,78 @@ class MovieSession(models.Model):
...
+        settings.AUTH_USER_MODEL,
+        on_delete=models.CASCADE,
+        null=True
+    )
+
+    class Meta:
+        ordering = ["-created_at"]
+
+    def __str__(self) -> str:
+        return f"{self.created_at}"
```
Comment (ArsenPidhoretskyi) [inline db/models.py:84]:
```suggestion
        return str(self.created_at)
```
In this case better to use str

File: services/user.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    return user
+
+
+def get_user(user_id: int) -> User:
+    return get_user_model().objects.get(id=user_id)
+
+
+def update_user(
+        user_id: int,
+        username: str = None,
```
Comment (ArsenPidhoretskyi) [inline services/user.py:36]:
```suggestion
        username: Optional[str] = None,
```
Because None as default. Fix in all places

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: db/models.py (hunk 3 above)
Comment (XOctOpus1) [reply db/models.py:84]: image

Comment (XOctOpus1) [reply db/models.py:84]: коли змінив з'явилась така помилка

Comment (XOctOpus1) [reply db/models.py:84]: а ні, то моя вина, забув забрати фігурні лапки

File: db/migrations/0002_user_order_ticket_alter_movie_actors_and_more.py [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('last_login', models.DateTimeField(blank=True, null=True, verbose_name='last login')),
+                ('is_superuser', models.BooleanField(default=False, help_text='Designates that this user has all permissions without explicitly assigning them.', verbose_name='superuser status')),
+                ('is_staff', models.BooleanField(default=False, help_text='Designates whether the user can log into this admin site.', verbose_name='staff status')),
+                ('is_active', models.BooleanField(default=True, help_text='Designates whether this user should be treated as active. Unselect this instead of deleting accounts.', verbose_name='active')),
+                ('date_joined', models.DateTimeField(default=django.utils.timezone.now, verbose_name='date joined')),
+                ('username', models.CharField(max_length=68, unique=True)),
+                ('password', models.CharField(max_length=68)),
+                ('email', models.EmailField(max_length=254)),
```
Comment (Yurnerroo) [inline db/migrations/0002_user_order_ticket_alter_movie_actors_and_more.py:29]: change ' to " with Black

**Review by Yurnerroo — CHANGES_REQUESTED** [review]

**Review by Yurnerroo — APPROVED** [review]

**Pull request closed without merging**